### PR TITLE
Support napari>=0.5

### PR DESCRIPTION
**Pull request by dstansby**
Solves 
- #287

### CODE REVIEW

**Comment by samcunliffe** [comment]
The tests fail because of this (significant) diff in the data ...

result-failed-diff

baseline

result

File: src/napari_matplotlib/tests/scatter/baseline/test_scatter_2D.png
Comment (samcunliffe) [inline src/napari_matplotlib/tests/scatter/baseline/test_scatter_2D.png:1]: Here also the axes are reversed.

**Comment by samcunliffe** [comment]
@dstansby I claim a fix here 🚀 

The 
```
OverflowError: Python integer -1 out of bounds for uint8
```
was fixed by -1  → -1.0 🙃 

I've somehow managed to push to your fork, so I can approve this, but I'll wait for **you** to click merge. 
(Also, we need to be sure to squash merge because there are images changed and changed back in the history here.)

**Review by samcunliffe — APPROVED** [review]

**Comment by samcunliffe** [comment]
pre-commit.ci autofix

**Comment by samcunliffe** [comment]
pre-commit.ci run

**Comment by dstansby** [comment]
I think the mypy error is because pre-commit is running a version of numpy (I can't work out which...) that doesn't have a `shape` argument to `np.reshape()`, which indeed seems only to have been added in 2.1.0: https://github.com/numpy/numpy/pull/26292. I think we want to support older numpy versions than that, so I think the fix is to revert to using the `newshape` argument, and ignoring the deprecation warning (for now, until we depend on numpy>=2.1)

**Comment by samcunliffe** [comment]
> so I think the fix is to revert to using the `newshape` argument, and ignoring the deprecation warning (for now, until we depend on numpy>=2.1)

Thanks for the detective work! I was just playing 🔨

**Pull request merged**